packages/liferay-npm-scripts/src/jest/mocks/liferay.js

```diff
@@ -0,0 +1,113 @@
+ * https://yuilibrary.com/yui/docs/api/classes/Attribute.html
+ */
+const events = {
+	/**
+	 * https://yuilibrary.com/yui/docs/api/files/event-custom_js_event-target.js.html#l372
+	 */
+	detach: jest.fn(),
+
+	/**
+	 * https://yuilibrary.com/yui/docs/api/files/event-custom_js_event-target.js.html#l700
+	 */
+	fire: jest.fn(),
+
+	/**
+	 * https://yuilibrary.com/yui/docs/api/files/event-custom_js_event-target.js.html#l227
+	 */
+	on: jest.fn(),
+
+	/**
+	 * https://yuilibrary.com/yui/docs/api/files/event-custom_js_event-target.js.html#l136
+	 */
+	once: jest.fn()
+};
+
+/**
+ * Contains a fallback/dummy implementation of `Liferay.Language.get`. In practice, this call is rewritten in a ServerFilter, so runtime calls to
+ * `Liferay.Language.get` should not be found in production code. A better match for the real behaviour would be a babel plugin to rewrite calls
+ * to the API with their "translated" value.
+ *
+ * https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/language.js
+ */
+const Language = {
+	/**
+	 * https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/language.js#L18
+	 */
+	get: jest.fn(key => key)
+};
+
+/**
+ * Contains APIs that provide information about the running context of the portal. The JS ThemeDisplay object is a representation of its Java counterpart
+ * simplified for JS access.
+ *
+ * https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/html/common/themes/top_js.jspf#L147
+ */
+const ThemeDisplay = {
+	/**
+	 * https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/html/common/themes/top_js.jspf#L217
+	 */
+	getDoAsUserIdEncoded: jest.fn(_ => 'default-mocked-do-as-user-id'),
+
+	/**
+	 * https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/html/common/themes/top_js.jspf#L235
+	 */
+	getPathMain: jest.fn(_ => '/c'),
+
+	/**
+	 * https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/html/common/themes/top_js.jspf#L238
+	 */
+	getPathThemeImages: jest.fn(_ => ''),
+
+	/**
+	 * https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/html/common/themes/top_js.jspf#L247
+	 */
+	getPortalURL: jest.fn(_ => 'http://localhost:8080')
+};
+
+/**
+ * General utilities on the `Liferay`. Possible API sources are:
+ *
+ * - https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js
+ * - https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js
+ */
+const Util = {
+	/**
+	 * https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js#L442
```

Comment:
All these links are doomed to become stale because what "master" points at will continuously update in the future; this is especially true of the ones that have line numbers, because the line numbers are likely to drift. (More likely than, say, a file getting renamed and a link turning into a 404).

I'd either link to a specific revision (press "y" when looking at the page you want to copy from on GitHub), or I'd drop the line numbers.

Reply:
argh, I knew this, because you’ve told us so many times... it’s just that my muscle memory hasn’t caught up with it yet.

I’ll recreate them